src/RestControllers/AuthorizationController.php

```diff
@@ -581,6 +586,204 @@ public function oauthAuthorizationFlow(): void
         }
     }
 
+    public function redirectToGoogle(AuthorizationRequest $authRequest): void
+    {
+        $client = $authRequest->getClient();
+        $this->logger->info(
+            "Redirecting to Google for GCIP authentication",
+            [
+                'client_id' => $client->getIdentifier(),
+            ]
+        );
+        $clientId = $client->getGoogleClientId();
+        $redirectUri = $this->authBaseFullUrl . '/callback/google';
+        $scope = implode(' ', array_map(fn($s) => $s->getIdentifier(), $authRequest->getScopes()));
+        $state = $authRequest->getState();
+        $nonce = $_SESSION['nonce'] ?? RandomGenUtils::produceRandomString(32);
+
+        $_SESSION['authRequestSerial'] = json_encode([
+            'outer' => [
+                'grantTypeId' => $authRequest->getGrantTypeId(),
+                'authorizationApproved' => false,
+                'redirectUri' => $authRequest->getRedirectUri(),
+                'state' => $authRequest->getState(),
+                'codeChallenge' => $authRequest->getCodeChallenge(),
+                'codeChallengeMethod' => $authRequest->getCodeChallengeMethod(),
+            ],
+            'scopes' => array_map(fn($s) => $s->getIdentifier(), $authRequest->getScopes()),
+            'client' => [
+                'name' => $client->getName(),
+                'redirectUri' => $client->getRedirectUri(),
+                'identifier' => $client->getIdentifier(),
+                'isConfidential' => $client->isConfidential(),
+            ],
+            'nonce' => $nonce,
+        ], JSON_THROW_ON_ERROR);
+
+        $authUrl = 'https://accounts.google.com/o/oauth2/v2/auth?' .
+            'response_type=code&' .
+            'client_id=' . urlencode($clientId) . '&' .
+            'redirect_uri=' . urlencode($redirectUri) . '&' .
+            'scope=' . urlencode($scope) . '&' .
+            'state=' . urlencode($state) . '&' .
+            'nonce=' . urlencode($nonce) . '&' .
+            'access_type=offline&prompt=consent';
+
+        header('Location: ' . $authUrl, true, 302);
+        exit;
```

Comment:
Exit statements like this are an antipattern -- this function should return something. I think you can follow the $response object setup as in https://github.com/msummers42/openemr/blob/8f14a00733a6c28a9dec0363bbd86e16d531a653/src/RestControllers/AuthorizationController.php#L462 and set `withHeader` and then `withStatus(302)` or whatever, and return the response.